flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/api/TableEnvironmentTest.scala

```diff
@@ -544,6 +545,38 @@ class TableEnvironmentTest extends TableTestBase {
       util.addTable[JTuple3[Int, Long, String]]('f0, 'f1.rowtime as 'new, 'f2),
       Seq("f0" -> INT, "new" -> ROWTIME, "f2" -> STRING))
   }
+
+  @Test
+  def testTableSchemaWithDifferentRowTypes(): Unit = {
+
+    def createInnerRow(innerFieldName: String): TypeInformation[_] = {
+      Types.ROW(
+        Array[String](innerFieldName),
+        Array[TypeInformation[_]](Types.INT()))
+    }
+
+    def createRow(innerFieldName: String): TypeInformation[_] = {
+      Types.ROW(
```

Comment:
If these 2 methods are only used by this test case, maybe we can eliminate them.

Reply:
These 2 methods are inner methods and used by multiple times. If we remove them, there will be some code duplication here.